### PR TITLE
Flag for non-linear integer arithmetic

### PR DESCRIPTION
**Pull request by Dspil**
Introduces a flag to disable non-linear integer arithmetic.

### CODE REVIEW

File: src/main/scala/viper/gobra/frontend/Config.scala [hunk 1]
```diff
@@ -646,6 +652,13 @@ class ScallopGobraConfig(arguments: Seq[String], isInputOptional: Boolean = fals
     noshort = true,
   )
 
+  val disableNL: ScallopOption[Boolean] = opt[Boolean](
+    name = "disableNL",
+    descr = "Disable non-linear integer arithmetics. Non compatible using Z3 via API or Carbon",
```
[the remark this answers is not in the document]
Comment (jcp19) [reply src/main/scala/viper/gobra/frontend/Config.scala:657]: Why doesn't it work with Z3 through API?

File: src/main/scala/viper/gobra/frontend/Config.scala [hunk 2]
```diff
@@ -770,6 +783,18 @@ class ScallopGobraConfig(arguments: Seq[String], isInputOptional: Boolean = fals
...
+  addValidation {
+    if (disableNL.toOption.contains(true)) {
+      if (z3APIMode.toOption.contains(true) || !isSiliconBasedBackend) {
+        Left("--disableNL is not compatible with Z3 via API or Carbon")
+      } else {
+        Right(())
+      }
+    } else {
+      Right(())
+    }
```
Comment (jcp19) [inline src/main/scala/viper/gobra/frontend/Config.scala:795]:
```suggestion
    if (disableNL.toOption.contains(true) && !z3APIMode.toOption.contains(true) && isSiliconBasedBackend) {
      Right(())      
    } else {
      Left("--disableNL is not compatible with Z3 via API or Carbon")
    }
```

File: src/test/resources/regressions/examples/disableNL.gobra [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+package pkg
+
+requires 0 <= x && x <= 10
+requires 0 <= y && y <= 10
+ensures 0 <= res && res <= 100 // would fail with --disableNL
```
Comment (jcp19) [inline src/test/resources/regressions/examples/disableNL.gobra:8]: Here, we can definitely add an annotation making this run with `--disableNL` and then we can add an expected post-condition failure

**Review by jcp19 — CHANGES_REQUESTED** [review]

File: src/main/scala/viper/gobra/frontend/Config.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dspil) [reply src/main/scala/viper/gobra/frontend/Config.scala:657]: We don't have a good way to pass these kinds of settings using the API.

File: src/main/scala/viper/gobra/frontend/Config.scala (hunk 2 above)
Comment (jcp19) [reply src/main/scala/viper/gobra/frontend/Config.scala:795]: oops, you are right, should be fixed now

**Review by jcp19 — APPROVED** [review]
LGTM

**Pull request merged**